api/python/provisioner/commands/deploy.py

```diff
@@ -268,35 +281,26 @@ def _run_states(self, states_group: str, run_args: run_args_type):
         primary = self._primary_id()
         secondaries = f"not {primary}"
 
+        hw_states = [
+            "system.storage.multipath",
+            "misc_pkgs.ipmi.bmc_watchdog",
+        ]
+
         # apply states
-        if setup_type == SetupType.SINGLE:
-            # TODO use salt orchestration
-            for state in states:
+        for state in states:
+
+            if state in hw_states and not self._is_hw():
+                continue
+
+            if setup_type == SetupType.SINGLE:
```

Comment:
We don't need specific sync states anymore.
Also, check if we can calculate setup type based on entries in command parameters or config.ini